pkg/controller/util/util.go

```diff
@@ -58,8 +58,7 @@ func Addresses(ctx context.Context, client client.Client) ([]string, error) {
 		return nil, err
 	}
 
-	addresses := make([]string, len(nodeList.Items))
-
+	var addresses []string
```

Comment:
You can still pre-allocate the memory for the slice knowing its capacity and then using `append` safely.

`addresses := make([]string, 0, len(nodeList.Items))` rather than

`addresses := make([]string, len(nodeList.Items))`.

In the first example, the size would be zero, so there won't be empty elements when you `append` to the slice. And that code would be more efficient because the runtime won't need to keep copying the data when it needs to grow the slice, since you'd already pre-allocated the memory by specifying the capacity via the third argument to `make`.

Reply:
So the only change you need to fix this is to add a 0 as the second argument to `make` and keep `len(nodeList.Items)` as the third.